capsules/src/signbus/app_layer.rs

```diff
@@ -0,0 +1,127 @@
+//! Kernel implementation of signbus_app_layer
+//! apps/libsignpost/signbus_app_layer.c -> kernel/tock/capsules/src/signbus_app_layer.rs
+//! By: Justin Hsieh
+//!
+//! Usage
+//! -----
+//!
+//! ```rust
+//! // Signbus App Layer
+//! let app_layer = static_init!(
+//!     capsules::signbus::app_layer::SignbusAppLayer<'static>,
+//!     capsules::signbus::app_layer::SignbusAppLayer::new(protocol_layer,
+//!             &mut capsules::signbus::app_layer::BUFFER0,
+//!             &mut capsules::signbus::app_layer::BUFFER1
+//! ));
+//!
+//! ```
+
+
+use core::cell::Cell;
+use kernel::ReturnCode;
+use kernel::common::take_cell::TakeCell;
+
+// Capsules
+use signbus::{protocol_layer, support, test_signbus_init};
+
+/// Buffers used to concatenate message information.
+pub static mut BUFFER0: [u8; 256] = [0; 256];
+pub static mut BUFFER1: [u8; 256] = [0; 256];
+
+/// SignbusAppLayer to handle userland interaction and adding message type information.
```

Comment:
Since this doesn't actually implement Driver (command, subscribe, etc.) this code doesn't actually directly interact with userland. Instead I'd say "to handle application messages"